parser_library/src/checking/ranged_diagnostic_collector.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2019 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+#include "ranged_diagnostic_collector.h"
+
+#include "diagnosable_impl.h"
+
+namespace hlasm_plugin::parser_library {
+
+ranged_diagnostic_collector::ranged_diagnostic_collector(const collectable<diagnostic_s>* diagnoser, range diag_range)
+    : s_diagnoser_(diagnoser)
+    , op_diagnoser_(nullptr)
+    , diag_range_(diag_range)
+    , diagnostics_present(false)
+{}
+
+ranged_diagnostic_collector::ranged_diagnostic_collector(const collectable<diagnostic_op>* diagnoser, range diag_range)
+    : s_diagnoser_(nullptr)
+    , op_diagnoser_(diagnoser)
+    , diag_range_(diag_range)
+    , diagnostics_present(false)
+{}
+
+ranged_diagnostic_collector::ranged_diagnostic_collector()
+    : s_diagnoser_(nullptr)
+    , op_diagnoser_(nullptr)
+    , diagnostics_present(false)
+{}
+
+void ranged_diagnostic_collector::operator()(const std::function<diagnostic_op(range)>& f) const
+{
+    diagnostics_present = true;
```

Comment:
this is confusing to me. why declare the function const when you intend to change the state of the object?

Reply:
The `add_diagnostic` method of the diagnosers is const and this method serves the same purpose, so I decided it should be const as well.
But it is true, it does not make sense. I will change it